src/main/java/Duke.java

```diff
@@ -1,10 +1,103 @@
+import java.text.ParseException;
+import java.util.Scanner;
+
+/**
+ * Represents a Duke chatbot.
+ */
 public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Initiates a Duke object.
+     * @param filePath path in hard disk to store tasks
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    private void done(int num) {
+        try {
+            tasks.done(num);
+            storage.write(tasks);
+            ui.done(tasks, num);
+        } catch (DukeException ex) {
+            ui.showDukeException(ex);
+        } catch (ParseException e) {
+            ui.showParseError();
+        }
+    }
+
+    private void delete(int num) {
+        try {
+            tasks.delete(num);
+            storage.write(tasks);
+            ui.delete(tasks, num);
+        } catch (DukeException ex) {
+            ui.showDukeException(ex);
+        } catch (ParseException e) {
+            ui.showParseError();
+        }
+    }
+
+    private void add(String type, String description) {
+        try{
+            tasks.add(type, description);
+            storage.write(tasks);
+            ui.add(tasks);
+        } catch (DukeException ex) {
+            ui.showDukeException(ex);
+        }
+    }
+
+    private void run() {
```

Comment:
could consider implementing the example run() given on the module website to make the run class more easy to read, especially since it contains the main logic of the entire program